spec/models/appointment_spec.rb

```diff
@@ -437,14 +437,40 @@
         end
       end
 
-      context 'appointment has been rescheduled' do
-        it 'needs another reminder' do
-          appointment.update!(
-            start_at: BusinessDays.from_now(20),
-            end_at:   BusinessDays.from_now(20) + 1.hour
-          )
+      context 'has already had a reminder' do
```

Comment:
I wonder whether those extra scenarios in the feature spec you added might be better expressed here instead... Given what they're ultimately testing is the inputs / outputs to the `needing_reminder` scope. Anyway, not overly concerned about this since the features are quick enough to run, but it's worth keeping an eye on if the logic grows any further perhaps.